core/src/test/java/org/arquillian/smart/testing/configuration/ConfigurationFileBuilder.java

```diff
@@ -7,44 +7,44 @@
 import org.yaml.snakeyaml.DumperOptions;
 import org.yaml.snakeyaml.Yaml;
 
-class ConfigurationFile {
+class ConfigurationFileBuilder {
 
     private HashMap<String, Object> properties;
 
-    private ConfigurationFile() {
+    private ConfigurationFileBuilder() {
         properties = new HashMap<>();
     }
 
-    static ConfigurationFile SmartTestingConfigurationFile() {
-        return new ConfigurationFile();
+    static ConfigurationFileBuilder SmartTestingConfigurationFile() {
```

Comment:
Uppercase in method name is a bit unusual for Java coders unless it's a constructor. Can we rename it to simple `configurationFile`? Also `create` could be `writeTo`, so it could be written as

```java
configurationFile()
    .mode("ordering")
    .applyTo("surefire")
    .inherit(Paths.get(root, SMART_TESTING_YAML).toString())
    .writeTo(Paths.get(root, SMART_TESTING_YML));
```